src/crc.rs

```diff
@@ -0,0 +1,39 @@
+//! CRC
+
+use crate::pac::CRC;
+use crate::rcc::{Enable, AHB};
+
+/// Extension trait to constrain the CRC peripheral
+pub trait CrcExt {
+    /// Constrains the CRC peripheral to play nicely with the other abstractions
+    fn new(self, ahb: &mut AHB) -> Crc;
+}
+
+impl CrcExt for CRC {
+    fn new(self, ahb: &mut AHB) -> Crc {
+        CRC::enable(ahb);
+        Crc { crc: self }
+    }
+}
+
+/// Constrained CRC peripheral
+pub struct Crc {
+    crc: CRC,
+}
+
+impl Crc {
+    pub fn read(&self) -> u32 {
+        self.crc.dr.read().bits()
+    }
+
+    pub fn write(&mut self, val: u32) {
+        unsafe { self.crc.dr.write(|w| w.bits(val)) }
```

Comment:
This `unsafe` is unneeded:
https://docs.rs/stm32f1/0.11.0/stm32f1/stm32f103/crc/dr/struct.DR_W.html

Reply:
Fixed, thanks!